src/main/java/com/redhat/rhjmc/containerjfr/net/web/WebServer.java

```diff
@@ -534,6 +542,25 @@ void handleAuthRequest(RoutingContext ctx) {
         }
     }
 
+    void handleHealthRequest(RoutingContext ctx) {
+        ctx.response().putHeader(HttpHeaders.CONTENT_TYPE, MIME_TYPE_JSON);
+        ctx.response().setStatusCode(200);
+
+        boolean datasourceAvailable = false;
+
+        if (this.env.hasEnv(GRAFANA_DATASOURCE_ENV)) {
+            HttpGet grafanaHealth = new HttpGet(this.env.getEnv(GRAFANA_DATASOURCE_ENV));
+            try {
+                CloseableHttpResponse response = this.client.execute(grafanaHealth);
+                if (response.getStatusLine().getStatusCode() == 200) {
+                    datasourceAvailable = true;
+                }
+            } catch (IOException e) {
+            }
+        }
+        endWithJsonKeyValue("datasourceAvailable", datasourceAvailable, ctx.response());
```

Comment:
When you do the refactor, this `endWithJsonKeyValue` isn't around anymore, so the preferred equivalent would look like `ctx.response().end(gson.toJson(Map.of("datasourceAvailable", datasourceAvailable)))`.

On that note, what do you think about adding an equivalent JSON property for the Grafana dashboard's health as well? It would be something like a `new HttpGet(this.env.getEnv(GRAFANA_DASHBOARD_ENV) + "/api/health"))`, and then the rest of the response processing you've done here would be the same I think.